firmware/application/database.cpp

```diff
@@ -56,6 +56,9 @@ int database::retrieve_aircraft_record(AircraftDBRecord* record, std::string sea
 }
 
 int database::retrieve_record(std::string file_path, int index_item_length, int record_length, void* record, std::string search_term) {
+    if (search_term.empty())
+        return DATABASE_RECORD_NOT_FOUND;
```

Comment:
Hi, yes I've been considering this. The reason why I didn't go with this solutions is cause it's kind of a lie :/ As in this case there is no query string, there is nothing to be found, the issue isnt that the record not found.

After thinking about it a couple more times, I'd rather go with a fourth enum elem, query error or something, but that should be checked outside anyhow, so even better an assert? :)